### DISCUSSION TITLE
Can we switch silently of DS3 pad ? Random input problem

### DISCUSSION THREAD

**Discussion by fraf** [nefarius/DsHidMini#146] [Q&A] [answered]
Hello,

I have played the emulator Ryujinx for one hour without problem. Then my sixaxis (first gen, not a DS3) ran out of battery. So I directly connected the other in bluetooth and the emulator give the command back again. But with the other one DS3 pad (a real DS3 this time), I have random inputs, it becomes unplayable. 
Can we use the same bluetooth dongle, the same installation of hidMini drivers in order to use another DS3 , so as to be able to switch from one DS3 to another when battery are low ?

Note : I tried to reset the DS3 (button in the back of the pad), to change from XInput to DS4Windows, but still random input under Ryujinx, whereas it was perfect the first time with a long play with the sixaxis initiallly.

Thanks.

**Comment by nefarius** [comment]
If I understood correctly, you wonder if you can connect many DS3s in parallel, wired or wireless, which yes of course you can do as much as you like.

As for random inputs, those come from the **hardware**, not the driver, the driver simply transaltes what it gets.

Cheers

**Reply by fraf** [reply]
OK thanks for your answer. I tried to uninstall all drivers+bluetooth. I reinstall all and my second controller (a real DS3) still have random inputs. That's very strange since it is a true DS3, does it means it is a fake one (I don't think so) or my windows problem ?
Very strange, the siaxis gamepad was perfect.
The DS3 is CECHZC2E A1 model.

**Reply by nefarius** [reply]
As I said, the driver works with whatever inputs it receives from the hardware, it rarely has anything to do with fake or genuine but more with worn out, damaged controllers like aged thumb sticks that start drifting etc.

**Comment by Kanuan** [comment]
>  to change from XInput to DS4Windows

1. Is DS4Windows detecting the controller when in DS4Windows mode? 
1. When using the controller with DS4Windows, how does the controller appears on the [Gamepad Tester Website](https://gamepad-tester.com/)? Does it responds as normal there?
    - Send a screenshot of how the controller appears in DS4Windows mode + DS4Windows software in the Gamepad Tester Website

**Reply by fraf** [reply]
1. Yes
2. Yes. Just the R3/L3 button which are not much responsive : if I click rapidly on both, sometimes there are misses (no highlight). Buttons + dpad, analog stick are responsive well.

**Comment by fraf** [comment, marked as answer]
So I tested a bit more. Appearently, the random input are less frequent in Wired mode than in Bluetooth mode. But I'am sorry it is not perfect input anyway. My reference is my Xbox One S gamepad. So I uninstalled all bluetooth/dshdmini drivers.
I reboot. I plug my xbox one wire first. then I paired it with my bluetooth dongle. All is perfect under Ryujinx (or whatever, that my test program). After that, I installed bluetooth drivers from wiki install process "install BthPS3 first". Then I continue and finish to install the drivers on my windows 10 : both the dualshock 3 + my xbox one s  controller are operationnal. I started ryujinx, choose my xbox one gamepad with snes online street fighter 2 rom : inputs are perfect. I pause the game, I switch input 1 to Nefarious controller (cf. scrrenshot : https://ibb.co/yhQGZ3g). I go back into the game, unpause and continue my play : fake input happens during battle ! For sure !.

Note : This problem is probably from my dualshock 3, I'am going to test same way back to my sixaxis controller.

**Reply by nefarius** [reply]
I can only repeat myself; *random inputs* come **from the controller (hardware)**, there is *nothing* the driver does to randomly generate inputs, it simply doesn't. Make sure the **hardware** is OK by for example using it with a PS3. You can re-install the driver a thousand times and the behaviour will not change, this is *not* a software issue.

**Reply by Kanuan** [reply]
From your screenshot, I'm assuming your controller is in ```Xinput``` mode.

As I said on the [other post you had previously created](https://github.com/ViGEm/DsHidMini/discussions/145#discussioncomment-1452355), the XInput Mode has some "issues" that require you launching games through Steam and having the 2 options instructed there enabled, **this includes launching Ryujinx itself through Steam**.

If you are on DS4Windows mode... then the screenshot shows that you are not actually in DS4Windows mode.

-----

**EDIT 1:** And as Nef said, DsHidMini just gets what the controller sends and translates it to the correct format accordingly to the selected HID Device Mode. Assuming you are using DsHidMini 100% correctly then whatever is happening is 99% chance of being the controller's fault

**Reply by fraf** [reply]
Thanks you all for your discussion. I played a bit more, and things seems to be returned to normal since I deleted Neferius drivers from add/delete program from windows panel (I did only the uninstall from DriverStoreExplorer  + uninstall on device manager). The faulty seems to be my windows (not well properly install/reinstalled drivers I would say). For information, with official sony drivers 3.0, RPCS3 was ok. Thanks for your answers guys anyway and your fast replies.
Cheers
